packages/components/src/site-thumbnail/use-dominant-color.tsx

```diff
@@ -0,0 +1,28 @@
+import FastAverageColor, { FastAverageColorResult } from 'fast-average-color';
+import { useEffect, useState } from 'react';
+
+export default function useDominantColor( url?: string ) {
+	const [ color, setColor ] = useState< FastAverageColorResult >();
+
+	useEffect( () => {
+		const fac = new FastAverageColor();
+
+		//get the dominant color of image from url
+		if ( url && url.length > 0 ) {
+			fac
+				.getColorAsync( url, {
+					ignoredColor: [
+						[ 255, 255, 255, 255 ], // white
+						[ 0, 0, 0, 255 ], // black
+					],
+					crossOrigin: 'anonymous',
+				} )
+				.then( ( color ) => setColor( color ) );
+		}
```

Comment:
So, how expensive this operation is? I think we can memoize it since the `url` is a valid hash and a collision is impossible...

The idea is to create a `Map` outside this hook (but in the same file) that would be queried in the initial value of `useState` and inside this effect. If found, bail out of the hook body since the value is already computed. If not, proceed with execution and store the result in the map.

Does it make sense?